README.md

```diff
@@ -145,104 +147,129 @@ Instead of *just* looking for `.goldengrahamsrc` (no extension), it will also lo
-explorer.search('start/search/here');
-explorer.search('start/search/at/this/file.css');
+- Starting from `process.cwd()` (or some other directory defined by the `searchPath` argument to [`search()`]), it looks for configuration objects in three places, in this order:
+  1. A `goldengrahams` property in a `package.json` file (or some other property defined by the cosmiconfig option [`packageProp`]);
+  2. A `.goldengrahamsrc` file with JSON or YAML syntax (or some other filename defined by the cosmiconfig option [`rc`]);
+  3. A `goldengrahams.config.js` JS file exporting the object (or some other filename defined by the cosmiconfig option [`js`]).
+- If none of those searches reveal a configuration object, it moves up one directory level and tries again. So the search continues in `./`, `../`, `../../`, `../../../`, etc., checking those three locations in each directory.
+- It continues searching until it arrives at your home directory (or some other directory defined by the cosmiconfig option [`stopDir`]).
+- If at any point a parseable configuration is found, the [`search()`] Promise resolves with its [result] \(or, in [`sync`] mode, the [result] is returned).
+- If no configuration object is found, the [`search()`] Promise resolves with `null` (or, in [`sync`] mode, `null` is returned).
+- If a configuration object is found *but is malformed* (causing a parsing error), the [`search()`] Promise rejects and shares that error (so you should `.catch()` it). (Or, in [`sync`] mode, the error is thrown.)
 
-// will stop on an empty configuration file
-explorer.search('start/search/here', { ignoreEmpty: false })
-```
+\*\*If you know where your configuration file should be located, you should instead use [`load()`], instead.
+
+#### searchPath
+
+Type: `string`.
+Default: `process.cwd()`.
 
-If you provide `searchPath`, cosmiconfig will start its search at `searchPath` and continue to search up the directory tree, as documented above.
-By default, `searchPath` is set to `process.cwd()`.
+[`search()`] will start its search from this directory.
 
-##### searchOptions
+#### searchOptions
 
-Can be an object containing a single property `ignoreEmpty`.  This defaults to true and tells cosmiconfig whether to continue searching
-for files when an empty one is found.
+##### ignoreEmpty
 
-#### `load([configPath])`
+Type: `boolean`.
+Default: `true`.
 
-Loads a configuration file directly.  Rejects with an error if the file does not exist, if the file contents are empty, or if the file cannot be parsed.  Upon success it returns a Promise that resolves with an object with two properties:
-- `config`: The loaded and parsed configuration.
-- `filepath`: The filepath where this configuration was found.
+By default, `search`, ignores empty configuration files and continues searching up the tree. If the `search` option [`ignoreEmpty`] is set to `false` and an empty configuration file is found, the [result] will have include `config: undefined` and `isEmpty: true`.
```

Comment:
I'm the worst with commas, but just checking to see if you intended on the comma after `search`

```txt
By default, `search`, ignores
                    ^
```